scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala

```diff
@@ -223,15 +223,35 @@ object Type {
     @replacedField("4.6.0") final def args: List[Type] = argClause.values
   }
   @ast class ApplyInfix(lhs: Type, op: Name, rhs: Type) extends Type
+  object FunctionType {
+    private[meta] def paramToParams(param: Type): List[Type] = param match {
+      case _: Lit.Unit => Nil
+      case Tuple((_: Tuple) :: Nil) => param :: Nil
+      case t: Tuple => t.args
+      case t => t :: Nil
+    }
+    private[meta] final val paramsToParam: List[Type] => Type = {
+      case Nil => Lit.Unit()
+      case t :: Nil if !t.is[Tuple] => t
+      case t => Tuple(t)
+    }
+  }
   @branch trait FunctionType extends Type {
-    def params: List[Type]
+    def param: Type
+    def params: List[Type] = FunctionType.paramToParams(param)
     def res: Type
   }
-  @ast class Function(params: List[Type], res: Type) extends FunctionType
+  @ast class Function(param: Type, res: Type) extends FunctionType {
+    @replacedField("4.6.0", FunctionType.paramsToParam) final override def params: List[Type] =
```

Comment:
Could you maybe add some docs about it and maybe add @deprecated flag? Otherwise users are getting both `param` and `params` when doing `.` completions on the tree node for Type.Function